content/tutorials/rust/hello-world.md

```diff
@@ -22,274 +24,127 @@ Before starting on this tutorial, you'll need to have completed the following
-futures = { version = "0.3.4", features = ['async-await'] }
-flv-future-aio = { version = "1.0.0" }
-flv-client = { version = "1.0.0"}
-kf-protocol = { version = "1.0.0 "}
+fluvio = "0.1.0"
+async-std = "1.0.0"
 ```
 
-
-##### Producer Code
-
-Add the following code in  _src/main.rs_ file:
+Now let's head on over to the `src/main.rs` file. This is where your `main` function
+lives. Since we'll be using async code, let's set up a helper function for `main` to
+call that will handle all of our async needs.
 
 ```rust
-use futures::stream::StreamExt;
-use flv_future_aio::task::run_block_on;
-use flv_client::{FluvioClient, ClientError};
+use fluvio::FluvioError;
 
 fn main() {
-    run_block_on(async {
-
-        match run_producer().await {
-            Ok(len) => println!("OK: {} bytes sent", len);
-            Err(err) => eprintln!("{}", err);
-        }
-
-    })
+    if let Err(error) = async_std::task::block_on(run()) {
+        println!("Error: {}", error);
+    }
 }
 
-async fn run_producer() -> Result<i32, ClientError> {
-    let fluvio_client = FluvioClient::new()?;
-    let mut connection = fluvio_client.connect().await?;
-    let mut replica = connection.get_replica("my-topic", 0).await?;
-    let len = replica.produce("test").await?;
-
-    Ok(len)
+async fn run() -> Result<(), FluvioError> {
+    todo!()
 }
 ```
 
-In summary:
-
-* *run_block_on* creates an async block
-* *FluvioClient::new()* reads the profile and creates a client.
-* *fluvio_client.connect()_ returns the connection to the cluster.
-* *connection.get_replica(...)* looks-up _replica_ for the topic/partition.
-* *replica.produce(...)* send a message to the _cluster_.
-
-##### Build and Run Producer
-
-Run _cargo build_ to generate binary:
-
-```bash
-$ cargo build
-   Compiling fluvio-rust-produce v0.1.0 (/Users/user/fluvio-rust-produce)
-    Finished dev [unoptimized + debuginfo] target(s) in 2.83s
-```
+This `run()` function is where we'll write the rest of our example code. Since it's an
+`async` function, that means that we can just use the `.await` keyword whenever we need
+to call another `async` function.
 
-Run _producer_ to send "test" to topic/partition _my-topic/0_ :
+Now let's write the code to produce (send) a message to Fluvio
 
-```bash
-./target/debug/fluvio-rust-produce
-OK: 4 bytes sent
+```rust
+async fn run() -> Result<(), FluvioError> {
+    let producer = fluvio::producer("hello-fluvio").await?;
+    producer.send_record("Hello, Fluvio!", 0);
+    Ok(())
+}
 ```
 
-#### Create Consumer Package
+We can already run this code and see it in action. We'll use the `fluvio` CLI to see
+the event arrive at the "hello-fluvio" topic.
 
-Create a consumer package in parallel with the producer.
+In one terminal window, run the following command to print out events in the "hello-fluvio"
+topic
 
 ```bash
-$ cd ..
-$ cargo new fluvio-rust-consume --bin
-Created binary (application) `fluvio-rust-consume` package
-$ cd fluvio-rust-consume/
+$ fluvio consume -B
```

Comment:
Hello Fluvio!

Reply:
At this point in the timeline, there would be no message. I'm telling them to open a terminal window that will be waiting to receive a message to they can see it arrive in real-time

Reply:
The previous sentence says that you can run this code, so assumed it would be done, hence my note.